lib/cfn-nag/custom_rules/EC2NetworkAclEntryProtocolRule.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require_relative 'base'
+
+class EC2NetworkAclEntryProtocolRule < BaseRule
+  def rule_text
+    'EC2 NetworkACL Entry Protocol must be either 6 (for TCP), 17 (for UDP), 1 (for ICMP), or 58' \
+    '(for ICMPv6, which must include an IPv6 CIDR block, ICMP type, and code).'
+  end
+
+  def rule_type
+    Violation::FAILING_VIOLATION
+  end
+
+  def rule_id
+    'F79'
+  end
+
+  def audit_impl(cfn_model)
+    violating_network_acl_entries = cfn_model.resources_by_type('AWS::EC2::NetworkAclEntry')
+                                             .select do |network_acl_entry|
+      violating_network_acl_entries?(network_acl_entry)
+    end
+
+    violating_network_acl_entries.map(&:logical_resource_id)
+  end
+
+  private
+
+  # https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateNetworkAclEntry.html#API_CreateNetworkAclEntry_RequestParameters
+  # A value of "-1" means all protocols. If you specify "-1" or a protocol
+  # number other than "6" (TCP), "17" (UDP), or "1" (ICMP), traffic on all ports
+  # is allowed, regardless of any ports or ICMP types or codes that you specify.
+  # If you specify protocol "58" (ICMPv6) and specify an IPv4 CIDR block,
+  # traffic for all ICMP types and codes allowed, regardless of any that you
+  # specify. If you specify protocol "58" (ICMPv6) and specify an IPv6 CIDR
+  # block, you must specify an ICMP type and code.
+
+  def tcp_udp_icmp_protocol?(network_acl_entry)
+    %w[1 6 17].include?(network_acl_entry.protocol)
+  end
+
+  def icmpv6_protocol?(network_acl_entry)
+    network_acl_entry.protocol == '58' && !network_acl_entry.ipv6CidrBlock.nil? &&
+      !network_acl_entry.icmp.nil? && !network_acl_entry.icmp['Code'].nil? &&
+      !network_acl_entry.icmp['Type'].nil?
+  end
+
+  def violating_network_acl_entries?(network_acl_entry)
+    if tcp_udp_icmp_protocol?(network_acl_entry) ||
+       icmpv6_protocol?(network_acl_entry)
```

Comment:
1. do we want to flag protocol == -1 as a warning?
2. wouldn't we want to consider this only when RuleAction == 'allow'?

Reply:
1.  Are you asking if we should only have `protocol == -1` as a warning or the entire `EC2NetworkAclEntryProtocolRule`?  This rule is currently a failing violation but I can see where it can be a warning as well since it this can be limited elsewhere.  In regards to being explicit about -1, I don't know that it is necessary since we are categorizing just the few protocols instead that would not trigger an opening of all ports.  The test scenario spec/test_templates/yaml/ec2_networkaclentry/ec2_networkaclentry_all_protocols.yml checks for this with a protocol of -1 and triggers the violation as it is now.
2.  The documentation states that anything that isn't one of the specific protocols will be _allowed_ on all ports.  I can play around with testing this a bit more to see how that scenario holds if set it to deny instead.

Reply:
1. ok makes sense.  -1 is covered alongside anything besides 1, 6, 17.
2. i like warning over fail because there are reason to allow all traffic in conjunction with security groups
3. i believe we only want to warn on Allows that are "open".  if something is set to Deny.... i don't think we warn to warn